### DISCUSSION TITLE
conditional classes don't get updated when element is hidden via x-show

### DISCUSSION THREAD

**Discussion by levu42** [livewire/livewire#4825] [7. Bugs (archived)]
```html
<div x-data="{editMode: false}">
  <button @class([
    'text-green' => is_null($title),
    'text-gray' => !is_null($title),
  ]) x-show="!editMode" @click="editMode = true">
    Edit
  </button>
  <div x-show="editMode">
    <input wire:model="title">
    <button wire:click="save()" @click="editMode = false">
  </div>
</div>
```

In the code above, the text-green / text-gray class gets updated, while the element is at the same time being shown again via x-show after having been hidden. Something in this combination makes that the change of class doesn't get applied to the button. I can see in the livewire debug tools, that the HTML returned form the server is correct and has the correct new class.

My temporary workaround is to put the `x-show` onto a temporary parent element of the element I apply class changes to:

```html
<div x-show="...">
  <button @class([...])>...</button>
</div>
```

**Comment by joshhanley** [comment]
@levu42 why don't you use Alpine to bind the classes to the data instead?

You can use `@entangle` to share the data with Alpine or use `$wire.title` directly to check which class to bind.

Hope this helps!
